src/software/world/timestamped_robot_state.h

```diff
@@ -0,0 +1,78 @@
+#pragma once
+
+#include "software/new_geom/angle.h"
+#include "software/new_geom/angular_velocity.h"
+#include "software/new_geom/point.h"
+#include "software/new_geom/vector.h"
+#include "software/time/timestamp.h"
+#include "software/world/robot_state.h"
+
+/**
+ * This class represents the state of a robot at a single point in time
+ */
+class TimestampedRobotState
+{
+   public:
+    /**
+     * Creates a new robot state with the given initial state and timestamp
+     *
+     * @param position The position of the robot, with coordinates in metres
+     * @param velocity The velocity of the robot, in metres per second
+     * @param orientation The orientation of the robot
```

Comment:
Maybe add radians for units here

Reply:
I actually removed the comment that specified radians because that's abstracted away in the `Angle` type